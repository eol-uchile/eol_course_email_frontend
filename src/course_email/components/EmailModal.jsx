import React from 'react'
import { Modal, Button } from '@edx/paragon';

export const EmailModal = ({ email, modalState, setModalState }) => {

    console.log('EmailModal loaded');

    const resetModalWrapperState = () => {
        setModalState(false);
    }

    const body = () => (
      <div className="row">
        <div className="font-weight-bold col col-3 py-2">
          Fecha
        </div>
        <div className="col col-9 py-2">
          { email.date }
        </div>  

        <div className="font-weight-bold col col-3 py-2">
          Enviado por
        </div>
        <div className="col col-9 py-2">
          { email.sender }
        </div> 

        <div className="font-weight-bold col col-3 py-2">
          Asunto
        </div>
        <div className="col col-9 py-2">
          { email.subject }
        </div> 

        <div className="font-weight-bold col col-3 py-2">
          Contenido
        </div>
        <div className="text-justify col col-9 py-2">
          { email.message }
        </div>
        
        { 
          email.receiver_users
          &&
          (
            <>
              <div className="font-weight-bold col col-3 py-2">
                Destinatarios ({ email.receiver_users.length })
              </div>
              <div className="text-justify col col-9 py-2">
                <ol>
                  {
                    email.receiver_users.map((user) => {
                      return(
                        <li key={ user }>
                          { user }
                        </li>
                      )
                    })
                  }
                </ol>
              </div>
            </>
          )
        }
      </div>
    );

    return (
        <Modal
          open={ modalState }
          title="Detalle del correo"
          body={ body() }
          onClose={ resetModalWrapperState }
          buttons={[
              <Button variant="primary" onClick={ resetModalWrapperState } data-autofocus >Cerrar</Button>
          ]}
          renderDefaultCloseButton={false}
          dialogClassName="modal-lg"
        />
    )
}
