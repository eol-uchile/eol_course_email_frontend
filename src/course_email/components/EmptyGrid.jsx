import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faFrown } from '@fortawesome/free-solid-svg-icons';

export const EmptyGrid = ( {title}) => {
    return (
        <div className="text-center">
            <FontAwesomeIcon icon={faFrown} className="h1" />
            <p>Al parecer no tienes { title.toLowerCase() }.</p>
        </div>
    )
}

EmptyGrid.propTypes = {
    title       : PropTypes.string.isRequired
}