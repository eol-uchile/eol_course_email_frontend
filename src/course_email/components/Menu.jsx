import React from 'react';
import { Button } from '@edx/paragon';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPaperPlane, faArrowDown, faArrowUp } from '@fortawesome/free-solid-svg-icons';

export default function Menu() {

    console.log('Menu loaded');

    return (
        <div>
            <div className="d-flex my-2">
                <Link to={`/eol/eol_course_email/static/received/`}>
                    <Button variant="outline-primary shadow-lg" size="sm"><FontAwesomeIcon icon={faArrowDown} className="mr-2" />Recibidos</Button>
                </Link>
                <Link to={`/eol/eol_course_email/static/sended/`} className="mr-auto">
                    <Button variant="outline-primary shadow-lg" size="sm" className="mx-3"><FontAwesomeIcon icon={faArrowUp} className="mr-2" />Enviados</Button>
                </Link>
                <Link to={`/eol/eol_course_email/static/new/`}>
                    <Button variant="primary shadow-lg" size="sm"><FontAwesomeIcon icon={faPaperPlane} className="mr-2" />Nuevo Correo</Button>
                </Link>
            </div>
            
        </div>
        
    )
}
