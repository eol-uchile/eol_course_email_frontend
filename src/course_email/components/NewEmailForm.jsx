import React from 'react';
import PropTypes from 'prop-types';
import { useFetchUsers } from '../hooks/useFetchUserData';
import { useForm } from '../hooks/useForm';
import { Spinner, Form, Button, Col } from '@edx/paragon';
import { StatusForm } from './StatusForm';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPaperPlane } from '@fortawesome/free-solid-svg-icons';


export const NewEmailForm = ( { courseId } ) => {
    const { users, loading } = useFetchUsers( courseId );
    const [ values, handleInputChange, handleSubmit] = useForm({
        status: "initialized",
        subjectInput    : "",
        messageInput   : "",
        studentsInput: [],
        staffInput: [],
        courseId: courseId
    })
    const { status, subjectInput, messageInput, studentsInput, staffInput } = values;

    return (
        <div className="rounded-lg shadow-lg py-4 px-5 my-2">
            <h3>Formulario Nuevo Correo </h3>
            <Form onSubmit={handleSubmit}>
                <Form.Row className="form-status">
                    <StatusForm status={status} />
                </Form.Row>
                <Form.Group controlId="formGridSubject">
                    <Form.Label className="lead">Asunto</Form.Label>
                    <Form.Control required placeholder="Asunto del correo" aria-describedby="subjectInput" name="subjectInput" onChange={ handleInputChange } value={subjectInput}/>
                </Form.Group>
                <Form.Group controlId="formGridMessage">
                    <Form.Label className="lead">Mensaje</Form.Label>
                    <Form.Control required as="textarea" placeholder="Mensaje del correo" aria-describedby="messageInput" name="messageInput" onChange={ handleInputChange } value={messageInput}/>
                </Form.Group>
                <hr/>
                <Form.Group id="formGridCheckbox">
                </Form.Group>
                <Form.Row>
                    { loading && <Spinner animation="border" variant="primary" className="d-flex mx-auto mt-2 "/> }
                    <Form.Group as={Col} controlId="formGridReceiverStudent" className="checkbox-group">
                        <Form.Label className="lead">Destinatarios 'Estudiantes'</Form.Label>
                        <div className="checkbox-input">
                            {
                                users
                                .filter( ({has_role}) => !has_role )
                                .map( (user) => (
                                    <Form.Check 
                                        type="checkbox" 
                                        className="pl-0"
                                        name="studentsInput"
                                        id={ user.username }
                                        key={ user.username }
                                        label={ user.name }
                                        onChange={ handleInputChange }
                                        checked={ studentsInput.includes(user.username) }
                                    />
                                ))
                            }
                        </div>
                        
                    </Form.Group>
                    
                    <Form.Group as={Col} controlId="formGridReceiverStaff" className="checkbox-group">
                        <Form.Label className="lead">Destinatarios 'Equipo docente'</Form.Label>
                        <div className="checkbox-input">
                            {
                                users
                                .filter( ({has_role}) => has_role )
                                .map( (user) => (
                                    <Form.Check 
                                        type="checkbox" 
                                        className="pl-0"
                                        name="staffInput"
                                        id={ user.username }
                                        key={ user.username }
                                        label={ user.name }
                                        onChange={ handleInputChange }
                                        checked={ staffInput.includes(user.username) }
                                    />
                                ))
                            }
                        </div>
                        
                    </Form.Group>
                </Form.Row>
                <Form.Row>
                    <Button variant="outline-primary shadow-lg" size="lg" type="submit" className="mx-auto" disabled={status=='pending'}>
                        <FontAwesomeIcon icon={faPaperPlane} className="mr-2" /> Enviar Correo
                    </Button>
                </Form.Row>
            </Form>
        </div>
    )
}

NewEmailForm.propTypes = {
    courseId       : PropTypes.string.isRequired
}