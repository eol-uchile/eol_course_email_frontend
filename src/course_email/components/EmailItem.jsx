import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { IconButton } from '@edx/paragon';
import { faInfo } from '@fortawesome/free-solid-svg-icons';
import { EmailModal } from './EmailModal';
export const EmailItem = ( { email } ) => {

    const [modalState, setModalState] = useState(false);

    const openModal = () => setModalState(true);

    return (
        <div className="row border-bottom">
            <div className="col col-3 pt-2">
                { email.date }
            </div>
                
            <div className="col col-3 pt-2 email-sender">
                { email.sender }
            </div>
                
            <div className="col col-5 pt-2">
                { email.subject }
            </div>
                
            <div className="col col-1">
                <IconButton icon={ faInfo } alt='Ver mayor informaciÃ³n del correo' onClick={ openModal } variant="primary"/>
            </div> 

            {
                modalState && <EmailModal email={email} modalState={modalState} setModalState={setModalState}/>
            }

        </div>
    )
}

EmailItem.propTypes = {
    email       : PropTypes.object.isRequired
}