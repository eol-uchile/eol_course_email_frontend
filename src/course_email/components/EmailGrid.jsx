import React from 'react';
import PropTypes from 'prop-types';
import { EmailItem } from './EmailItem';
import { Spinner } from '@edx/paragon';
import { useParams } from 'react-router';
import { useFetchEmails } from '../hooks/useFetchUserData';

export const EmailGrid = ( { getEmails, title } ) => {
    const { courseId } = useParams();
    console.log(`${title} loaded`);
    const { data, loading } = useFetchEmails( courseId, getEmails );
    return (
        <div className="rounded-lg shadow-lg py-4 px-5 my-2">
            <h3>{title}</h3>
            { loading && <Spinner animation="border" variant="primary" className="d-flex mx-auto mt-2 "/> }
            <div className="email-grid">
                <div className="row font-weight-bold border-bottom py-2">
                    <div className="col col-3">
                        Fecha
                    </div>
                    <div className="col col-3">
                        Enviado por
                    </div>
                        
                    <div className="col col-5">
                        Asunto
                    </div>
                        
                    <div className="col col-1 text-center">
                        Detalle
                    </div> 
                </div>
                {
                    data.map( (email, index) => (
                        <EmailItem 
                            key={ index }
                            email= { email }
                        />
                    ))
                }
            </div>
        </div>
    )
}

EmailGrid.propTypes = {
    getEmails   : PropTypes.func.isRequired,
    title       : PropTypes.string.isRequired
}