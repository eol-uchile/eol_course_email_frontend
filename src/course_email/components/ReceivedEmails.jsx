import React from 'react';
import PropTypes from 'prop-types';
import { useFetchReceivedEmails } from '../hooks/useFetchUserData';
import { EmailItem } from './EmailItem';
import { Spinner } from '@edx/paragon';

export const ReceivedEmails = ({ courseId }) => {
    const { data, loading } = useFetchReceivedEmails( courseId );
    return (
        <>
            <h1>Emails Recibidos</h1>
            { loading && <Spinner animation="border" variant="primary" /> }
            <div className="email-grid">
                <div className="row font-weight-bold border-bottom py-2">
                    <div className="col col-3">
                        Fecha
                    </div>
                    <div className="col col-3">
                        Enviado por
                    </div>
                        
                    <div className="col col-5">
                        Asunto
                    </div>
                        
                    <div className="col col-1 text-center">
                        Detalle
                    </div> 
                </div>
                {
                    data.map( (email, index) => (
                        <EmailItem 
                            key={ index }
                            { ...email }
                        />
                    ))
                }
            </div>
        </>
    )
}

ReceivedEmails.propTypes = {
    courseId : PropTypes.string.isRequired,
}