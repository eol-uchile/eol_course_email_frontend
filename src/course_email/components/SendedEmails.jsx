import React from 'react';
import { useFetchSendedEmails } from '../hooks/useFetchUserData';
import { EmailItem } from './EmailItem';
import { Spinner } from '@edx/paragon';
import { useParams } from 'react-router';

export const SendedEmails = () => {
    const { courseId } = useParams();
    console.log('SendedEmails loaded');
    const { data, loading } = useFetchSendedEmails( courseId );
    return (
        <div className="rounded-lg shadow-lg py-4 px-5 my-2">
            <h3>Correos Enviados</h3>
            { loading && <Spinner animation="border" variant="primary" className="d-flex mx-auto mt-2 "/> }
            <div className="email-grid">
                <div className="row font-weight-bold border-bottom py-2">
                    <div className="col col-3">
                        Fecha
                    </div>
                    <div className="col col-3">
                        Enviado por
                    </div>
                        
                    <div className="col col-5">
                        Asunto
                    </div>
                        
                    <div className="col col-1 text-center">
                        Detalle
                    </div> 
                </div>
                {
                    data.map( (email, index) => (
                        <EmailItem 
                            key={ index }
                            { ...email }
                        />
                    ))
                }
            </div>
        </div>
    )
}