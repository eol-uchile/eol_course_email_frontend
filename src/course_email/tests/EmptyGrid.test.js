import React from 'react';
import { render, screen } from './TestUtils';
import { EmptyGrid } from '../components/EmptyGrid';

describe('Testing <EmptyGrid /> App Component', () => {
    const title = "correos recibidos";
    let wrapper = render(<EmptyGrid title={title} />);
    
    // initialization before each test
    beforeEach( () =>  {
        wrapper = render(<EmptyGrid title={title} />);
    });

    test('Load <EmptyGrid /> correctly', () => {
        
        expect( wrapper ).toMatchSnapshot();

    });

    test('Show message correctly', () => {

        expect(screen.getAllByText(`Al parecer no tienes ${title}.`));

    })
    
})