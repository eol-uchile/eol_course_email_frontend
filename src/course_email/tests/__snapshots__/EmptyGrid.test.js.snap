// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <EmptyGrid /> App Component Load <EmptyGrid /> correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-center"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-frown fa-w-16 h1"
          data-icon="frown"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 496 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm80 168c17.7 0 32 14.3 32 32s-14.3 32-32 32-32-14.3-32-32 14.3-32 32-32zm-160 0c17.7 0 32 14.3 32 32s-14.3 32-32 32-32-14.3-32-32 14.3-32 32-32zm170.2 218.2C315.8 367.4 282.9 352 248 352s-67.8 15.4-90.2 42.2c-13.5 16.3-38.1-4.2-24.6-20.5C161.7 339.6 203.6 320 248 320s86.3 19.6 114.7 53.8c13.6 16.2-11 36.7-24.5 20.4z"
            fill="currentColor"
          />
        </svg>
        <p>
          Al parecer no tienes 
          correos recibidos
          .
        </p>
      </div>
    </div>
    <div>
      <div
        class="text-center"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-frown fa-w-16 h1"
          data-icon="frown"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 496 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm80 168c17.7 0 32 14.3 32 32s-14.3 32-32 32-32-14.3-32-32 14.3-32 32-32zm-160 0c17.7 0 32 14.3 32 32s-14.3 32-32 32-32-14.3-32-32 14.3-32 32-32zm170.2 218.2C315.8 367.4 282.9 352 248 352s-67.8 15.4-90.2 42.2c-13.5 16.3-38.1-4.2-24.6-20.5C161.7 339.6 203.6 320 248 320s86.3 19.6 114.7 53.8c13.6 16.2-11 36.7-24.5 20.4z"
            fill="currentColor"
          />
        </svg>
        <p>
          Al parecer no tienes 
          correos recibidos
          .
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-center"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-frown fa-w-16 h1"
        data-icon="frown"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 496 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm80 168c17.7 0 32 14.3 32 32s-14.3 32-32 32-32-14.3-32-32 14.3-32 32-32zm-160 0c17.7 0 32 14.3 32 32s-14.3 32-32 32-32-14.3-32-32 14.3-32 32-32zm170.2 218.2C315.8 367.4 282.9 352 248 352s-67.8 15.4-90.2 42.2c-13.5 16.3-38.1-4.2-24.6-20.5C161.7 339.6 203.6 320 248 320s86.3 19.6 114.7 53.8c13.6 16.2-11 36.7-24.5 20.4z"
          fill="currentColor"
        />
      </svg>
      <p>
        Al parecer no tienes 
        correos recibidos
        .
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
